tests/main/refresh-devmode-nonproduction/task.yaml

```diff
@@ -0,0 +1,30 @@
+summary: Check that the refresh command works.
+
+systems: [-ubuntu-core-16-64, -ubuntu-core-16-device-amd64, -ubuntu-core-16-device-arm]
+
+details: |
+    These tests exercise the refresh command using different store backends.
+    The concrete store to be used is controlled with the STORE_TYPE environment
+    vairiable, if it is empty, a fake local store is used, if it has a value of
+    staginig or production the corresponding remote store is used.
+    When executing against the remote stores the tests rely in the existence of
+    a given snap with an updatable version (version string like 2.0+fake1) in the
+    edge channel.
+
+environment:
+    SNAP_NAME: test-snapd-tools
+    SNAP_VERSION_PATTERN: \d+\.\d+\+fake1
+    BLOB_DIR: $(pwd)/fake-store-blobdir
+    STORE_TYPE: fake
```

Comment:
This is related to the prior point about multiple systems which are in fact the same device. The distinction is not in fact a different system, or even a different task. The task is exactly the same, on the same system, and indeed the difference lies in that we want variations of that test. That's what variants are for, as you point out, and I think they remain a better way to implement this test.

You can very easily exclude a particular setup on particular conditions by doing
```
if [ something ]; then exit; fi